utils/icinga2_client.go

```diff
@@ -35,6 +36,39 @@ func NewIcinga2Client(address string, username string, password string) *Icinga2
 	}
 }
 
+// getNoBody successfully returns http.NoBody.
+func getNoBody() (io.ReadCloser, error) {
+	return http.NoBody, nil
+}
+
+func (c *Icinga2Client) Do(req *http.Request) (*http.Response, error) {
+	if req.Body == nil && req.GetBody == nil {
+		req.Body = http.NoBody
+		req.GetBody = getNoBody
+	}
+
+	for attempt := 1; ; attempt++ {
+		response, err := c.Client.Do(req)
+		if err == nil && response.StatusCode == http.StatusServiceUnavailable && req.GetBody != nil && attempt < 300 {
```

Comment:
`attempt < 300` with `attempt` starting at 1 results in performing `c.Client.Do()` 300 times, not re-trying 300 times.

Reply:
The number of attempts isn't even the most relevant part here. It retries for roughly 5 minutes, which should be plenty for our tests.